compose.go

```diff
@@ -184,3 +184,10 @@ func (cs *ComposeStore[V]) ClearAll(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (cs *ComposeStore[V]) GetOrSetWithLock(
+	ctx context.Context, key string, getter func(context.Context, string) (V, error),
+) (V, error) {
+	var out V
+	return out, fmt.Errorf("cachestore: GetOrSetWithLock is unsupported on ComposeStore")
+}
```

Comment:
Not 100% sure what the behaviour should be in the case of ComposeStore... the simplest (and probably most sane?) option could be to only use the last store (as the last one is usually slowest but most reliable, e.g. most likely Redis)...